boa/src/builtins/number/mod.rs

```diff
@@ -383,7 +383,11 @@ impl Number {
 
         // 2. If radix is undefined, let radixNumber be 10.
         // 3. Else, let radixNumber be ? ToInteger(radix).
-        let radix = args.get(0).map_or(10, |arg| arg.to_integer()) as u8;
+        let radix = args
+            .get(0)
+            .map(|arg| arg.to_integer(ctx))
+            .transpose()?
+            .unwrap_or(10.0) as u8;
```

Comment:
I would change this to do a `map(|radix| radix as u8).unwrap_or(10);` That way we would avoid a conversion from `{float}` to `u8` in most cases.